agent/main.go

```diff
@@ -246,34 +246,40 @@ func handleBootstrapFlow(logger logr.Logger, hostName string) error {
 func certificateRotation(logger logr.Logger, hostName string, config *rest.Config) error {
 	var pollDuration = 5 * time.Second
 	for {
-		block, _ := pem.Decode(config.CertData)
-		if block == nil || block.Type != "CERTIFICATE" {
-			logger.Info("failed to decode PEM block containing certificate")
-			return nil
-		}
-
-		cert, err := x509.ParseCertificate(block.Bytes)
-		if err != nil {
-			logger.Error(err, "Certifcate parse failed")
+		if err := certRotation(logger, hostName, config); err != nil {
 			return err
 		}
+		// Poll after every few seconds
+		time.Sleep(pollDuration)
+	}
+}
 
-		totalTimeCert := cert.NotAfter.Sub(cert.NotBefore)
+func certRotation(logger logr.Logger, hostName string, config *rest.Config) error {
```

Comment:
Suggestion: We have extracted all of the code except for the loop in this function. Could you see if it's okay to move the for loop [here](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/agent/main.go#L179)?

Reply:
As the main method is already handling a lot of things, to avoid more complexity it would be better to keep the for loop in a method of its own.